src/libostree/ostree-sysroot-deploy.c

```diff
@@ -2082,6 +2082,47 @@ install_deployment_kernel (OstreeSysroot   *sysroot,
   g_autofree char *options_key = ostree_kernel_args_to_string (kargs);
   ostree_bootconfig_parser_set (bootconfig, "options", options_key);
 
+  g_autoptr(GError) local_error = NULL;
+  GKeyFile *config = ostree_repo_get_config (repo);
+  g_autofree char *read_value = g_key_file_get_string (config, "sysroot", "bls-append-except-default", &local_error);
+  /* We can ignore not found errors */
+  gboolean not_found = g_error_matches (local_error, G_KEY_FILE_ERROR, G_KEY_FILE_ERROR_KEY_NOT_FOUND) || \
+                       g_error_matches (local_error, G_KEY_FILE_ERROR, G_KEY_FILE_ERROR_GROUP_NOT_FOUND);
+  if (not_found) 
+    {
+      g_clear_error(&local_error);
+    } 
+  else if (local_error) 
+    {
+      g_propagate_error (error, g_steal_pointer (&local_error));
+      return FALSE;
+    }
+
+  /* Only append to this BLS config if:
+   * - this is not the default deployment
+   */
+   /* If deployment was prepended, it is the new default */
+  gboolean is_new_default = ostree_deployment_get_index(deployment) == 0;
```

Comment:
Personally like wrapping these in parentheses for clarity. E.g.

```suggestion
  gboolean is_new_default = (ostree_deployment_get_index (deployment) == 0);
```